tools/integration_tests/local_file/stat_file_test.go

```diff
@@ -0,0 +1,132 @@
+	}
+}
+
+func assertStatCallFileSize(expected, got int64, t *testing.T) {
+	if got != expected {
+		t.Errorf("File size mismatch in stat call. Expected: %d, Got: %d", expected, got)
+	}
+}
+
+func assertStatCallFileMode(got os.FileMode, t *testing.T) {
+	if got != FilePerms {
+		t.Errorf("File permissions mismatch in stat call. Expected: %v, Got: %v", FilePerms, got)
+	}
+}
+
+// //////////////////////////////////////////////////////////////////////
+// Tests
+// //////////////////////////////////////////////////////////////////////
+func TestStatOnLocalFile(t *testing.T) {
+	// Clean the mountedDirectory before running test.
+	setup.CleanMntDir()
+	// Create a local file.
+	filePath, fh := CreateLocalFile(FileName1, t)
+
+	// Stat the local file.
+	fi, err := os.Stat(filePath)
+	assertStatCallErrorIsNil(err, t)
+	assertStatCallFileName(path.Base(filePath), fi.Name(), t)
+	assertStatCallFileSize(0, fi.Size(), t)
+	assertStatCallFileMode(fi.Mode(), t)
+
+	// Writing contents to local file shouldn't create file on GCS.
+	WritingToLocalFileShouldNotWriteToGCS(fh, FileName1, t)
+
+	// Stat the local file again to check if new content is written.
+	fi, err = os.Stat(filePath)
+	assertStatCallErrorIsNil(err, t)
+	assertStatCallFileName(path.Base(filePath), fi.Name(), t)
+	assertStatCallFileSize(10, fi.Size(), t)
+	assertStatCallFileMode(fi.Mode(), t)
+
+	// Close the file and validate if the file is created on GCS.
+	CloseFileAndValidateObjectContents(fh, FileName1, FileContents, t)
+}
+
+func TestStatOnLocalFileWithConflictingFileNameSuffix(t *testing.T) {
+	// Clean the mountedDirectory before running test.
+	setup.CleanMntDir()
+	// Create a local file.
+	filePath, fh := CreateLocalFile(FileName1, t)
+	// Stat the local file.
+	fi, err := os.Stat(filePath + inode.ConflictingFileNameSuffix)
+	assertStatCallErrorIsNil(err, t)
+	assertStatCallFileName(path.Base(filePath)+inode.ConflictingFileNameSuffix, fi.Name(), t)
+	assertStatCallFileSize(0, fi.Size(), t)
+	assertStatCallFileMode(fi.Mode(), t)
+
+	// Close the file and validate if the file is created on GCS.
+	CloseFileAndValidateObjectContents(fh, FileName1, "", t)
+}
+
+func TestTruncateLocalFile(t *testing.T) {
+	// Clean the mountedDirectory before running test.
+	setup.CleanMntDir()
+	// Create a local file.
+	filePath, fh := CreateLocalFile(FileName1, t)
+	// Writing contents to local file .
+	WritingToLocalFileShouldNotWriteToGCS(fh, FileName1, t)
+
+	// Stat the file to validate if new contents are written.
+	fi, err := os.Stat(filePath)
+	assertStatCallErrorIsNil(err, t)
+	assertStatCallFileName(path.Base(filePath), fi.Name(), t)
+	assertStatCallFileSize(10, fi.Size(), t)
+	assertStatCallFileMode(fi.Mode(), t)
+
+	// Truncate the file to update the file size.
+	err = os.Truncate(filePath, 5)
+	if err != nil {
+		t.Errorf("os.Truncate err: %v", err)
+	}
+	ValidateObjectNotFoundErr(FileName1, t)
+
+	// Stat the file to validate if file is truncated correctly.
+	fi, err = os.Stat(filePath)
```

Comment:
Why dont you put entire stat and assert logic in one method and pass filePath and expected filesize as arguments.